From 3a4062b40be0df37aed34565119a44f1218c7a08 Mon Sep 17 00:00:00 2001
From: wangzhengyong <wangzhengyong@cmss.chinamobile.com>
Date: Mon, 21 Nov 2016 17:32:55 +0800
Subject: [PATCH] sheepdog: support external IO NIC function

---
 block/sheepdog.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/block/sheepdog.c b/block/sheepdog.c
index 74f65c0..8675042 100644
--- a/block/sheepdog.c
+++ b/block/sheepdog.c
@@ -131,6 +131,8 @@ struct node_id {
     uint16_t port;
     uint8_t io_addr[16];
     uint16_t io_port;
+    uint8_t ext_io_addr[16];
+    uint16_t ext_io_port;
     uint8_t pad[4];
 };
 
@@ -1533,8 +1535,8 @@ static int sd_get_nodes(BDRVSheepdogState *s, Error **errp)
         for (i = 0; i < nr_nodes; i++) {
             struct sheep_host *p = s->hosts + i;
             p->port = nodes[i].nid.port;
-            if (!inet_ntop(AF_INET, nodes[i].nid.addr + 12,
-                p->addr, INET_ADDRSTRLEN)) {
+	    if (!inet_ntop(AF_INET, nodes[i].nid.ext_io_port? nodes[i].nid.ext_io_addr + 12
+			:nodes[i].nid.addr + 12, p->addr, INET_ADDRSTRLEN)) {
                 g_free(s->hosts);
                 ret = SD_RES_SYSTEM_ERROR;
                 goto out;
-- 
1.8.3.1

